src/controlobjectscript.cpp

```diff
@@ -0,0 +1,50 @@
+#include <QApplication>
+#include <QtDebug>
+
+#include "controlobjectscript.h"
+
+ControlObjectScript::ControlObjectScript(const ConfigKey& key, QObject* pParent)
+        : ControlObjectSlave(key, pParent) {
+}
+
+void ControlObjectScript::connectScriptFunction(
+        const ControllerEngineConnection& conn) {
+    if (m_connectedScriptFunctions.isEmpty()) {
+        // we connect the slots only, if there will be actually a script
+        // connected
+        connect(m_pControl.data(), SIGNAL(valueChanged(double, QObject*)),
+                this, SLOT(slotValueChanged(double,QObject*)),
+                Qt::QueuedConnection);
+        connect(this, SIGNAL(trigger(double, QObject*)),
+                this, SLOT(slotValueChanged(double,QObject*)),
+                Qt::QueuedConnection);
+    }
+    m_connectedScriptFunctions.append(conn);
+}
+
+bool ControlObjectScript::disconnectScriptFunction(
+        const ControllerEngineConnection& conn) {
+    bool ret = m_connectedScriptFunctions.removeAll(conn) > 0;
+    if (m_connectedScriptFunctions.isEmpty()) {
+            // no script left, we can disconnected
+            disconnect(m_pControl.data(), SIGNAL(valueChanged(double, QObject*)),
+                    this, SLOT(slotValueChanged(double,QObject*)));
+            disconnect(this, SIGNAL(trigger(double, QObject*)),
+                    this, SLOT(slotValueChanged(double,QObject*)));
+    }
+    return ret;
+}
+
+void ControlObjectScript::slotValueChanged(double value, QObject*) {
+    for(auto conn: m_connectedScriptFunctions) {
```

Comment:
I have no experiences with it. From http://en.cppreference.com/w/cpp/language/range-for
It looks like we should use for(auto&& .. to avoid a copy. Do I got it right?